extensions/scrollback-3.3.md

```diff
@@ -0,0 +1,70 @@
+
+Alternatively, content can be requested using a `draft/msgid`:
+
+    SCROLLBACK target draft/msgid=ID message_count
+    SCROLLBACK #channel draft/msgid=774ba1b6-202b-448c-b23a-6150ce5681fd 50
+
+If no message_count is known, `*` should be used to specify the default value.
+
+## Examples
+The examples below are written with `draft/msgid` tags included. This tag is optional but recommended.
+
+### Begin
+    :irc.host BATCH +ID chathistory target
+    :irc.host BATCH +XNyDSitp9MvcX scrollback #channel
+### PRIVMSG
+    @batch=ID;draft/msgid=UUID;time=YYYY-MM-DDThh:mm:ss.sssZ :nick!ident@host PRIVMSG target :message
+    @batch=XNyDSitp9MvcX;draft/msgid=eb703092-0782-4c28-bf7c-f9e5c45963ca;time=2016-11-19T18:02:01.001Z :foo!bar@example.com PRIVMSG #channel :The SCROLLBACK specification is going to be fantastic!
+### NOTICE
+    @batch=ID;draft/msgid=UUID;time=YYYY-MM-DDThh:mm:ss.sssZ :nick!ident@host NTOICE target :message
+    @batch=XNyDSitp9MvcX;draft/msgid=cb302078-ce4e-4866-82b8-480ec58094b3;time=2016-11-19T18:02:02.002Z :foo!bar@example.com NOTICE #channel :Announcing the new IRCv3 SCROLLBACK extension draft!
+### End
+    :irc.host BATCH -ID
+    :irc.host BATCH -XNyDSitp9MvcX
+
+## Use cases
+The batch type and supporting command are useful for allowing an "infinite scroll" type capability within the client itself. A client will, upon scrolling to the top of the active window, request scrollback from the server and, upon receiving such content, append it to the top of the window. Users can repeat this historic scrolling to retrieve prior history until limitations are met (see below).
+
+## Limitations
+Logging of messages and other actions must be enabled server-side and can be stored in any format desired, given appropriate software exists to retrieve and format such stored contnet. Scrollback can only be retrieved as far back as logs exist for the requesting user in the specified channel or query.
+
+A method for securely identifying the requesting user must exist to ensure content is sent only to the appropriate users and clients. See below for more information.
+
+## Security considerations
+Secure identification of users and clients must exist in order to ensure that users cannot obtain history that does not belong to them. Use of account names, internal account identifiers, or certificate fingerprints should be strongly considered when matching content to users. The server must verify the current user's identity matches that of the desired content. This information is not sent as part of the `SCROLLBACK` command and must be validated via other means, such as those stated above.
+
+## Current Implementations
+A ZNC-module implementation exists as [znc-scrollback](https://github.com/MuffinMedic/znc-scrollback). Client-side support for the ZNC module is in progress. Interest in supporting the batch type and command without ZNC has also been obtained, although specifics are not available at this time.
+
+[batch]: batch-3.2.md
```

Comment:
This should link to the website pages for those specs rather than the markdown documents.